File: src/main/java/de/cplaiz/activecraftcore/utils/config/Feature.java
package de.cplaiz.activecraftcore.utils.config;

public enum Feature {

    // TODO: 05.08.2022 was geht zu features machen

    PREFIX("prefix");

    private final String identifier;

    Feature(String identifier) {
        this.identifier = identifier;
    }

    public String identifier() {
        return identifier;
    }

    public static Feature fromIdentifier(String identifier) {
        for (Feature feature : values()) {
            if (feature.identifier.equals(identifier)) {
                return feature;
            }
        }
        return null;
    }
}

File: src/main/java/org/activecraft/activecraftcore/utils/config/Feature.kt
package org.activecraft.activecraftcore.utils.config

enum class Feature(private val identifier: String) {

    // TODO: 05.08.2022 was geht zu features machen
    PREFIX("prefix");

    fun identifier() = identifier

    companion object {
        fun fromIdentifier(identifier: String): Feature? {
            return values().find { it.identifier == identifier }
        }
    }
}